src/main/java/com/learning/pages/PageBase.java

```diff
@@ -1,14 +1,16 @@
 package com.learning.pages;
 
+import com.learning.browsers.Browser;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.PageFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public abstract class PageBase {
 
   protected final WebDriver driver;
 
-  public PageBase(WebDriver driver) {
-    this.driver = driver;
+  public PageBase(@Autowired Browser browser) {
```

Comment:
Not sure If adding @Autowired to parent abstract class is correct.

Reply:
Yes it is, but you can autowire to field directly, this way all paged that inherit from it won't have to pass browser downstream.